src/Safe7579Launchpad.sol

```diff
@@ -227,48 +230,36 @@ contract Safe7579Launchpad is IAccount, SafeStorage, ISafeOp {
         // ensure that the call was successful
         if (!success) revert InvalidUserOperationData();
 
-        // to support validation with the safe native signers, we allow to set the validator module
-        // to address(0)
-        if (validator == address(0)) {
+        // Call onInstall on each validator module to set up the validators.
+        // Since this function is delegatecalled by the SafeProxy, the Validator Module is
+        // called
+        // with msg.sender == SafeProxy.
+        bool userOpValidatorInstalled;
+        uint256 validatorsLength = initData.validators.length;
+        for (uint256 i; i < validatorsLength; i++) {
+            address validatorModule = initData.validators[i].module;
+            IValidator(validatorModule).onInstall(initData.validators[i].initData);
+            emit ModuleInstalled(MODULE_TYPE_VALIDATOR, validatorModule);
+
+            if (validatorModule == validator) userOpValidatorInstalled = true;
+        }
+
+        // Ensure that the validator module selected in the userOp was
+        // part of the validators in InitData
+        if (userOpValidatorInstalled) {
+            // validate userOp with selected validation module.
+            validationData = IValidator(validator).validateUserOp(userOp, userOpHash);
+        } else {
```

Comment:
note: this is now same behavior as safe7579 where it falls back to safe not just on zero nonce key but on any nonce key that is not an installed validator - is this still what we want?

Reply:
I think for launchpad zero nonce key is sufficient?!